client/signup/steps/domains/index.jsx

```diff
@@ -244,12 +245,25 @@ export class RenderDomainsStep extends Component {
 		}
 	};
 
-	handleDomainMappingError = ( domain_name ) => {
+	handleDomainMappingError = async ( domain_name ) => {
 		const productToRemove = this.props.cart.products.find(
 			( product ) => product.meta === domain_name
 		);
+
 		if ( productToRemove ) {
-			this.removeDomain( { domain_name, product_slug: productToRemove.product_slug } );
+			this.setState( ( prevState ) => ( {
+				domainRemovalQueue: [
+					...prevState.domainRemovalQueue,
+					{ meta: productToRemove.meta, productSlug: productToRemove.product_slug },
+				],
+			} ) );
+			this.setState( { isValidatingDomains: true } );
+			await this.removeDomain( { domain_name, product_slug: productToRemove.product_slug } );
+			this.setState( { isValidatingDomains: false } );
```

Comment:
Should this be gated behind the queue being empty?

Reply:
After a couple of tests, the result is the same, since we're awaiting that return. Thanks for the suggestion.